libs/utils/src/lib/docs-versions.tsx

```diff
@@ -2,31 +2,46 @@
  * ********************************************
  * ** List of versions of the documentation  **
  * ********************************************
- *
- * For each version, you must specify 2 things:
- * Label - We will use this for the sidebar.
- * ID - We will use it to create folders.
- * Branch, tag or commit - Used to fetch data.
- *
  **/
 
-export interface DocsVersion {
+/**
+ * Produces an object type that must contain exactly one key (in union K) of type V
+ * See: https://stackoverflow.com/a/60873215
+ */
+type ExactlyOneKeyWithType<K extends keyof any, V, KK extends keyof any = K> =
+  { [P in K]: { [Q in P]: V } &
+    { [Q in Exclude<KK, P>]?: never} extends infer O ?
+    { [Q in keyof O]: O[Q] } : never
+  }[K];
+
+type BaseDocsVersion = {
+  /** Used for visual display, e.g. the docs sidebar */
   label: string;
+  /** Used for routes */
   id: string;
-  branch?: string;
-  tag?: string;
-  commit?: string;
+  /** Used in calculations */
+  number: number;
+  isPreRelease?: boolean;
 }
```

Comment:
I'm a bit confused at what's happening here. In no circumstances we will update this array in the app so I'm wondering why adding more complexity to it. Looking at the type definition, I'm just a bit lost.

Reply:
Actually, another change made these benefits moot. I had to pull the actual `docsVersions` definition into a `/docs-version.js` file (which had to be CJS) at the project root, so that I could require it inside a CJS module for next-sitemap's config. Then I import that definition into this file, apply the type, and re-export it.

This complex typing was only to aid the maintenance of the `docsVersions` definition. But because of the CJS shenanigans, we cannot actually type that definition anymore, so there's no benefit.

So, long story, short... I removed all this complexity.